src/process_utility.c

```diff
@@ -870,16 +870,27 @@ process_index_end(Node *parsetree, CollectedCommand *cmd)
 	bool		handled = false;
 
 	Assert(IsA(stmt, IndexStmt));
-	Assert(cmd->type == SCT_Simple);
 
 	hcache = hypertable_cache_pin();
 	ht = hypertable_cache_get_entry_rv(hcache, stmt->relation);
 
 	if (NULL != ht)
 	{
+		ObjectAddress obj;
+
+		switch (cmd->type)
+		{
+		    case SCT_Simple:
+				obj =  cmd->d.simple.address;
+				break;
+		    default:
+				elog(ERROR, "Not yet supported on hypertables");
```

Comment:
Is it possible to figure out under which circumstances this happens so that we know throwing an error won't break things?

Reply:
I'll just remove it for consistency.